lib/playground.dart

```diff
@@ -453,6 +458,13 @@ class Playground implements GistContainer, GistController {
       }));
 
     _context = new PlaygroundContext(editor);
+    _context.onModeChange.listen((_) {
+      if (_context.activeMode == "dart") {
```

Comment:
For terseness, this could be `formatButton.disabled = _context.activeMode != 'dart'`